src/NServiceBus.ContainerTests/When_using_nested_containers.cs

```diff
@@ -135,8 +113,41 @@ public void Instance_per_call_components_should_not_cause_memory_leaks()
                 var upperLimitBytes = 200*1024;
                 Assert.That(after - before, Is.LessThan(upperLimitBytes), "Apparently {0} consumed more than {1} KB of memory", builder, upperLimitBytes/1024);
             }
+        }
+
+        [Test]
+        [Explicit]
+        public void Instance_per_uow_components_should_not_cause_memory_leaks()
```

Comment:
> UOW instances in child containers are singletons from the parent containers

I think that's not true. Classes which are registered in UOW scope ON the parent container should be singletons within the child container. The instance returned from the parent and the one returned from the child container are not the same.